Update to maintained Ruby version 2.7.6

agrberg opened the pull request:
~Built on top of an intended to be merged after https://github.com/dblock/iex-ruby-client/pull/113~

Now built on top of https://github.com/dblock/iex-ruby-client/pull/117 to restore CI and then update to a maintained Ruby.

resolves #115

dblock reviewed (changes requested):
I don't see many benefits to making Ruby 2.7 the minimum. Maybe we can keep it at 2.4? 2.6? for now? Either way we need CI back to see what's possible #116

agrberg commented on .rubocop.yml:
```diff
@@ -15,7 +17,11 @@ Naming/FileName:
 Style/Documentation:
   Enabled: false
 
+Layout/LineLength:
```
I agree it's definitely gotten in my way before. The [reasoning in the style guide](https://rubystyle.guide/#max-line-length) and the [additional cops this helps with](https://docs.rubocop.org/rubocop/1.32/cops_layout.html#layoutlinelength) are convincing.

As it currently passes w/ this "fix" for lines ending in long comments, would it be acceptable to wait and see how long this is good enough for?

dblock replied:
I'd prefer we disabled it instead of trying to be clever, but it's NBD.

agrberg replied:
Ultimately I didn't feel strongly enough and made the simpler change. I've definitely confused my future-self by being clever in the past 🙃

agrberg commented on iex-ruby-client.gemspec:
```diff
@@ -8,10 +8,9 @@ Gem::Specification.new do |s|
-  s.required_rubygems_version = '>= 1.3.6'
-  s.required_ruby_version = '>= 2.3.0'
+  s.required_rubygems_version = '~> 3.0'
+  s.required_ruby_version = '~> 2.7.6'
```
🤦 yes

agrberg commented on iex-ruby-client.gemspec:
```diff
@@ -8,10 +8,9 @@ Gem::Specification.new do |s|
+  s.required_rubygems_version = '~> 3.0'
+  s.required_ruby_version = '~> 2.7.6'
   s.files = `git ls-files`.split("\n")
-  s.test_files = `git ls-files -- spec/*`.split("\n")
```
This field was deprecated from Gemspecs and Rubocop's [`Gemspec/DeprecatedAttributeAssignment`](https://docs.rubocop.org/rubocop/1.32/cops_gemspec.html#gemspecdeprecatedattributeassignment) got it

agrberg commented on iex-ruby-client.gemspec:
```diff
@@ -20,9 +19,10 @@ Gem::Specification.new do |s|
-  s.add_development_dependency 'rake', '~> 10'
+  s.add_development_dependency 'rake'
   s.add_development_dependency 'rspec'
-  s.add_development_dependency 'rubocop', '0.72.0'
```
Ah! Good point. Would it be better to lock it to the current latest version on this branch or let it float w/ [NewCops](https://github.com/dblock/iex-ruby-client/pull/114/commits/c90c558ad1b99181ba07c46030bd21d5327a2c04#diff-4f894049af3375c2bd4e608f546f8d4a0eed95464efcdea850993200db9fef5cR4) disabled? Failing CI would encourage any minor changes from Rubocop updates to be made as opposed to whole new cops.

dblock replied:
I'd lock it with the most recent version possible and `NewCops: enable`, again just not trying to be too clever with it.

agrberg commented:
> I don't see many benefits to making Ruby 2.7 the minimum. Maybe we can keep it at 2.4? 2.6? for now? Either way we need CI back to see what's possible #116

Good call regarding CI. It wasn't straightforward but it ended up being some minimal changes. I rebased this on #117 and split up the commits into an incredibly minimal [730dd06](https://github.com/dblock/iex-ruby-client/pull/114/commits/730dd062af7d73519a05ebc69b645411347dcb6f) to update the Ruby version.

Now [ec4020f](https://github.com/dblock/iex-ruby-client/pull/114/commits/ec4020f59c15899160dfac95fd0ef3e64bdd6c11) contains the safer changes and rubocop todo updates as well as additional current locks on the dev dependencies.

dblock commented:
Thanks so much for all of this! Let's get the CI to pass and cleanup the rest of the PRs.

dblock commented:
Rebase this one and let's re-review?

agrberg commented:
This one is ready but it still depends on the question of whether or not we want to restrict the library to only maintained versions of Ruby. The Rubocop updates are nice but hardly required.

dblock commented:
> This one is ready but it still depends on the question of whether or not we want to restrict the library to only maintained versions of Ruby. The Rubocop updates are nice but hardly required.

I think I'd rather not restrict the library unnecessarily, it's bound to make someone unhappy, I'm still on 2.6.x for example in at least one project that uses this. WDYT 

> FWIW It looks like the token I copied from your Slack client project [doesn't have write access](https://github.com/dblock/iex-ruby-client/runs/7473374501?check_suite_focus=true#step:4:11) to the project. While the output is not familiar, it appears to be successful and simply unable to state that it is successful via a PR comment.

I opened https://github.com/dblock/iex-ruby-client/issues/118, we can deal with it later. Danger isn't supposed to need write access, it just needs to be able to comment on the PRs, which is something any account (and therefore this super limited PAT) can do.

agrberg commented:
I agree. Going to close this. We'll revisit minimum Ruby version if there's a forward facing problem in the future like all the kwarg changes for Ruby 3.
